src/responder/common/responder_common.c

```diff
@@ -194,7 +194,7 @@ errno_t check_allowed_uids(uid_t uid, size_t allowed_uids_count,
 }
 
 errno_t csv_string_to_uid_array(TALLOC_CTX *mem_ctx, const char *csv_string,
-                                bool allow_sss_loop,
+                                bool prevent_sss_loops,
```

Comment:
Having this variable named `prevent_sss_loops` in c-file and `allow_sss_loop` in a header is a bit confusing.

Besides, it looks it is always `false` outside tests. I would consider removing this arg altogether.

Reply:
Btw, do we really want to allow sssd resolvable users in ifp/pac configs?

Reply:
> Having this variable named `prevent_sss_loops` in c-file and `allow_sss_loop` in a header is a bit confusing.

Right. Will fix it.

Reply:
Those three functions executing this function with ` allow_sss_loop = true` were tests. So maybe they need to be adapted because they were actually testing a situation that would never happen in real life.
I agree that we need to make a decision at this point before proceeding.

Reply:
> Those three functions executing this function with ` allow_sss_loop = true` were tests.

No. Check current sources:
```
$ grep -rn csv_string_to_uid_array *
responder/pam/pamsrv.c:72:        ret = csv_string_to_uid_array(pctx->rctx, uid_str, true,
responder/pac/pacsrv.c:93:    ret = csv_string_to_uid_array(pac_ctx->rctx, uid_str, true,
responder/ifp/ifpsrv.c:213:    ret = csv_string_to_uid_array(ifp_ctx->rctx, uid_str, true,
```

Reply:
So @sumit-bose's comment is to keep original behavior intact, i.e.to **allow** `_SSS_LOOPS` for this function for all 3 responders.
Currently patches changes this behavior for sssd_pam and this needs to be fixed.

Reply:
I absolutely agree. I was convinced I have just removed 'true' parameters from tests. I will add that parameter back.

Reply:
Well, first patch inverted meaning of parameter, so 2nd patch indeed dropped 'true' from tests only.

But that's not the point.

The point is that now different responders have different env and in some cases this doesn't match our needs in `csv_string_to_uid_array()`.
Btw, this doesn't mean we need to keep additional parameter. This means we just should make sure loops are allowed inside this function and env restored back at exit.

Reply:
> So, if I understood correctly, you are raising a flag because ifp's and pac's behaviors are being changed.What I need to do is call `server_setup(..., allow_sss_loop=false)` for those two components to retrieve the previous behavior. Do you agree?

No.

`server_setup(allow_sss_loop)` invocations are correct.

We are discussing `csv_string_to_uid_array()` only.

Reply:
In all three functions you mentioned `csv_string_to_uid_array()` was being called with `allow_sss_loop=true` and now they are being called with 'prevent_sss_loops=false'. I fail to see the change in behavior there.

Reply:
- original state: `csv_string_to_uid_array()` always does `unsetenv("_SSS_LOOPS")`
 - after patches in this PR: `csv_string_to_uid_array()` runs with env of calling process (that is different pam vs pac/ifp)

I described this in https://github.com/SSSD/sssd/pull/6133#discussion_r898949502

Reply:
After a discussion on IRC, I understand now what's wrong. I'll fix it soon. Thanks.

Reply:
@justin-stephenson The last commit removes the parameter you asked me to rename. Could you please check that you agree with this change?